src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -146,6 +156,7 @@ public RubyContext(RubyLanguage language, TruffleLanguage.Env env) {
         // We need to construct this at runtime
         random = new SecureRandom();
 
+        // TODO (eregon, 25 Jan. 2018): This seed is made constant by context pre-initialization.
         final long hashingSeed;
```

Comment:
This is a pretty important `TODO` - please don't forget about it.

Reply:
Is this `TODO` still valid? A new `SecureRandom` is created and assigned when the context is reinitialised, and random is set to null when saving the image.

Reply:
Yes, the `Hashing` instance captures this seed and is not reset.
And it cannot be reset! Otherwise keys created during initialization would hash differently than keys after.
This seems hard to solve.
That seed is used for internal hashtables, not for Ruby-level `#hash` though.